### PR TITLE
[DEV-3970] Notes Styles Touch-Up

### PR DESCRIPTION
**Pull request by cdanwards**
### What does this PR do, or why is it needed?

Just some styling touchups in the notes feature.

### How do I test this PR?

Fire up a sim and go to the 'Freedom from the inside out' piece of content.

## TODO

- [ ] I am affirming this is my _best_ work ([Ecclesiastes 9:10](https://www.bible.com/bible/97/ECC.9.10.MSG))
- [ ] PR has a relevant title that will be understandable in a public changelog (ie...non developers)
- [ ] No new warnings in tests, in storybook, and in-app
- [ ] Upload GIF(s) of iOS and Android if applicable
- [ ] Set a relevant reviewer

## REVIEW

- [ ] Review updates to test coverage and snapshots
- [ ] Review code through the lens of being concise, simple, and well-documented

**Manual QA**

- [ ] Manual QA on iOS and ensure it looks/behaves as expected
- [ ] Manual QA on Android and ensure it looks/behaves as expected

---

> The purpose of PR Review is to _improve the quality of the software._

### CODE REVIEW

**Comment by cdanwards** [comment]
@richarddubay @redreceipt  - One of the snapshots from the locations was updated when I updated the snaps... Not sure if that should have happened. Just a note.

File: packages/newspringchurchapp/src/content-single/Features/TextFeature.js [hunk 1]
```diff
@@ -13,10 +13,11 @@ const TextFeature = ({
   card,
 }) => {
   const [isPressed, press] = useState(false);
-  const bodyWithBlank = body.replace(/__(.*)__/gm, (match, p1) =>
+  const bodyWithComma = body.replace(/%2C/, ',');
```
Comment (redreceipt) [inline packages/newspringchurchapp/src/content-single/Features/TextFeature.js]: this should be caught and handled on the back end

Comment (redreceipt) [reply packages/newspringchurchapp/src/content-single/Features/TextFeature.js]: ...in core

File: packages/newspringchurchapp/src/content-single/WeekendContentItem/index.js [hunk 2]
```diff
@@ -36,37 +41,48 @@ const WeekendContentItem = ({ content, loading }) => {
...
-                        {content.title}
-                      </H2>
-                      <HTMLContent contentId={content.id} />
-                    </ThemeMixin>
-                  </Header>
-                  <MediaControls contentId={content.id} />
-                  <Features contentId={content.id} asNotes />
-                  <HorizontalContentFeed contentId={content.id} />
-                </FlexedScrollView>
+                <FlexedAvoidantView
```
Comment (redreceipt) [inline packages/newspringchurchapp/src/content-single/WeekendContentItem/index.js]: what is this for? I hate that we're deviating from core so much on this component

File: packages/newspringchurchapp/src/content-single/Features/NoteFeature.js [hunk 3]
```diff
@@ -1,17 +1,34 @@
...
+import {
+  ActionCard,
+  Icon,
+  TextInput,
+  styled,
+  Touchable,
+} from '@apollosproject/ui-kit';
 import ShareContentButtonConnected from 'newspringchurchapp/src/ui/ShareContentButtonConnected';
 
+const StyledTextInput = styled(({ theme }) => ({
```
Comment (redreceipt) [inline packages/newspringchurchapp/src/content-single/Features/NoteFeature.js:14]: 🔥 🔥 🔥

File: packages/newspringchurchapp/src/user-settings/Locations/__snapshots__/Locations.tests.js.snap [hunk 4]
```diff
@@ -107,41 +107,28 @@ exports[`Display Native Map with Locations Render a native map view 1`] = `
                       }
                     }
                   >
-                    <View
-                      animate={null}
-                      customAnimate={null}
-                      onReady={false}
+                    <Image
```
Comment (redreceipt) [inline packages/newspringchurchapp/src/user-settings/Locations/__snapshots__/Locations.tests.js.snap]: yea this file shouldn't have changed, something fishy is happening here

File: packages/newspringchurchapp/src/content-single/WeekendContentItem/index.js (hunk 2 above)
Comment (cdanwards) [reply packages/newspringchurchapp/src/content-single/WeekendContentItem/index.js]: This makes the text input adjust for the keyboard. Core won't have this because they're not doing free form notes.

Comment (redreceipt) [reply packages/newspringchurchapp/src/content-single/WeekendContentItem/index.js]: Ok so then I think our signature in this file should be as small as possible, instead of adding all this here, simply replace the `ScrollView` above with a custom component containing `AvoidingView` and `ScrollView` with all your props in a different file. So something like this:

```jsx
21:    const FlexedScrollView = styled({ flex: 1 })(CustomAvoidingScrollView);
```

Where `CustomAvoidingScrollView` comes from a different file entirely in the same directory. What do you think?

File: packages/newspringchurchapp/src/content-single/Features/TextFeature.js (hunk 1 above)
Comment (cdanwards) [reply packages/newspringchurchapp/src/content-single/Features/TextFeature.js]: Would we not just do the same thing in the backend? If so, does it really matter where we do this? I'm of the opinion that it doesn't.

Comment (redreceipt) [reply packages/newspringchurchapp/src/content-single/Features/TextFeature.js]: it's a bug in core that the message data field of a text feature returns text that isn't parsed correctly, that should be fixed.

Comment (redreceipt) [reply packages/newspringchurchapp/src/content-single/Features/TextFeature.js]: we would have to repeat this code everywhere on the front we used a text feature in the future. technically this same thing is broken in core on the front end right now, why would we not fix it there?

Comment (cdanwards) [reply packages/newspringchurchapp/src/content-single/Features/TextFeature.js]: I guess I had the assumption that the text feature wouldn't be included outside of a WeekendContentItem? I'm honestly not sure. We can PR this against core though.

**Comment by redreceipt** [comment]
Core has WeekendContentItems too. I still think it should be on the back
end though, when you do the core work, you should put the code there.

--
Michael Neeley
Junior Web Developer
NewSpring Church
www.newspring.cc
Cell: (803) 622-6599

File: packages/newspringchurchapp/src/user-settings/Locations/__snapshots__/Locations.tests.js.snap (hunk 4 above)
Comment (cdanwards) [reply packages/newspringchurchapp/src/user-settings/Locations/__snapshots__/Locations.tests.js.snap]: @redreceipt @richarddubay My local snaps are messed up somehow. Y'all should run the snaps and update this branch.

**Review by redreceipt — DISMISSED** [review]
move the comma check out of this one and into a core PR and I think it's good to go!

File: packages/newspringchurchapp/src/content-single/Features/TextFeature.js (hunk 1 above)
Comment (richarddubay) [reply packages/newspringchurchapp/src/content-single/Features/TextFeature.js]: So yeah, a `TextFeature` could technically be used in multiple places, not just the `WeekendContentItem`. I think I agree that this should probably be fixed in core. And I guess I want to know if there is a way to parse the text so that you don't have to do `bodyWithComma` and `bodyWithQuote` and `bodyWith...`, you get the idea. But that can be figured out as this task gets taken on. Parsing all of the text at once would be ideal.

**Comment by richarddubay** [comment]
Screenshot 2019-09-03 16 50 47

@cdanwards I think we can lose some of this extra space too. There's just a lot of space that doesn't have to be there. Just give it the same amount of spacing that the rest of the items have.

**Comment by redreceipt** [comment]
I looked into that one time, think it's gonna be a little complicated but
definitely doable, there's built in bottom padding to the paragraphs on the
scripture renderer. Gonna be a core fix.

--
Michael Neeley
Junior Web Developer
NewSpring Church
www.newspring.cc
Cell: (803) 622-6599

**Comment by richarddubay** [comment]
I want to make sure that we can get whatever we can into our release on Thursday. @redreceipt Do you think this is something that we can fix in both places? I don't really like doing that, but we're running close on time.

**Comment by redreceipt** [comment]
Not by the solution I'm thinking of but maybe there's another way.

--
Michael Neeley
Junior Web Developer
NewSpring Church
www.newspring.cc
Cell: (803) 622-6599

**Comment by cdanwards** [comment]
@richarddubay I think that might have to be a core fix. I'm looking at the scripture renderer and it doesn't look straightforward.

**Comment by richarddubay** [comment]
@cdanwards @redreceipt Would it look different if the last thing before a note wasn't scripture? That's what I think we need to fight against ... and test.

File: packages/newspringchurchapp/src/content-single/Features/TextFeature.js (hunk 1 above)
Comment (cdanwards) [reply packages/newspringchurchapp/src/content-single/Features/TextFeature.js]: I'm making a PR against Core for this to be handled on the backend. I think we should leave this in until that gets merged.

Comment (redreceipt) [reply packages/newspringchurchapp/src/content-single/Features/TextFeature.js]: I disagree, we likely won't even have notes before it gets merged and even if we do, just make sure there are no commas

Comment (richarddubay) [reply packages/newspringchurchapp/src/content-single/Features/TextFeature.js]: Are we sure that it's only commas that are at issue here? Are there other punctuation marks or the like that would/could/are causing problems?

Comment (redreceipt) [reply packages/newspringchurchapp/src/content-single/Features/TextFeature.js]: Commas are all I'm aware but I'm sure there's more. That's why this particular fix really doesn't serve us very well.

Comment (richarddubay) [reply packages/newspringchurchapp/src/content-single/Features/TextFeature.js]: Yeah. I'm not sure off the top of my head how to do it, but whatever the fix is should parse all the special characters correctly.

Comment (richarddubay) [reply packages/newspringchurchapp/src/content-single/Features/TextFeature.js]: @cdanwards I think I'm in agreement with @redreceipt here. Let's take out the comma fix (since I really feel like the fix should be wider ranging than just a comma) and we can then merge this in. I want to make sure we get this in our new beta release tomorrow. We'll make sure on our end that the notes we do have don't include anything breaking until the core fix is done and in our codebase. Feel free to make an issue in core and then fix that issue. Go team, go!

File: packages/newspringchurchapp/src/content-single/WeekendContentItem/index.js [hunk 5]
```diff
@@ -1,5 +1,5 @@
 import React from 'react';
-import { ScrollView, Dimensions } from 'react-native';
+import { Dimensions } from 'react-native';
```
Comment (redreceipt) [inline packages/newspringchurchapp/src/content-single/WeekendContentItem/index.js:2]: why is this here?

Comment (redreceipt) [reply packages/newspringchurchapp/src/content-single/WeekendContentItem/index.js:2]: sike I see now

**Comment by richarddubay** [comment]
So a few things for posterity: 

1. The conversation in this PR has been awesome. This is how PR's should be. 
2. I moved the issue for the spacing after the scripture feature to its own Jira issue. Do we know for sure this is a core issue? Is there extra spacing in their scripture feature cards?
3. I'm good to approve this now.

**Review by richarddubay — APPROVED** [review]

**Comment by cdanwards** [comment]
@richarddubay 
1. 👍 
2. I'm not sure that this is an issue. Core never intended for the notes to occupy the same card. We'll just have to do something to gain the control to edit the styles - or just not put a `PaddedView` after Scripture Features like we do every other one.
3. Approve away
